File: 001/solution_001.ml
let rec project_euler_001 n acc =
  match n with
      0 -> acc
    | n ->
      match (n mod 3 == 0) || (n mod 5 == 0) with
          true ->
            project_euler_001 (n - 1) (n + acc)
        | false ->
            project_euler_001 (n - 1) (acc)


let project_euler_001 n =
  let solution = project_euler_001 (n - 1) 0 in
    print_int solution;
    print_endline ""


let () = project_euler_001 1000

File: 001/solution_001.hs
condition :: Int -> Bool
condition x = mod x 3 == 0 || mod x 5 == 0

solution :: Int -> Int
solution n =
    sum $ [x | x <- [1 .. n - 1], condition x]

project_euler_001 :: Int -> IO ()
project_euler_001 n =
    print $ solution n

main :: IO ()
main = project_euler_001 1000
